CMakeLists.txt

```diff
@@ -65,7 +65,16 @@ else()
     )
 endif()
 
-set(CMAKE_BUILD_TYPE Release)
+set(default_build_type "Release")
+ 
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
+  set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE
+      STRING "Choose the type of build." FORCE)
```

Comment:
Do you really need to cache this value as CMAKE_BUILD_TYPE is already cached?

Reply:
I got this pattern from here: https://blog.kitware.com/cmake-and-the-default-build-type/
Kitware are the makers of CMake